.github/workflows/ci.yml

```
@@ -0,0 +1,368 @@
+name: ci
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+
+env:
+  GRADLE_OPTS: "-Xmx4g -Dorg.gradle.daemon=false"
+  total-runners: 8
+
+jobs:
+  compile:
+    runs-on: ubuntu-latest
+    container: tmio/besu-build:1.0
+    timeout-minutes: 30
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v3
+      - name: Set up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: adopt
+          java-version: 11
```

I think this is the right place to upgrade to Java 17 now.
```suggestion
          java-version: 17
```

There are several places to do this.

---

that's out of scope. I don't mind indulging you, but if there's any breakage, it needs to go into a subsequent PR.